src/cocotb_bus/compat.py

```diff
@@ -0,0 +1,34 @@
+# Copyright cocotb contributors
+# Licensed under the Revised BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-3-Clause
+
+
+from packaging.version import parse as parse_version
+import warnings
+
+import cocotb
+
+def TestFactory(*args, **kwargs):
+    with warnings.catch_warnings():
+        # TestFactory has been deprecated in v2.0 and there's no migration path in v1.x
+        warnings.simplefilter("ignore")
+        return cocotb.regression.TestFactory(*args, **kwargs)
+
+
+# Note that we want to not use coroutine in beta versions of cocotb 2.0 too.
+# Therefore we cannot use 2.0 in comparison, because
+# parse_version("2.0a1") < parse_version("2.0")
+if parse_version(cocotb.__version__) > parse_version("1.999.0"):
```

Comment:
I think you can use `2.dev0`? If not don't worry about it.

Reply:
Thanks, this is much cleaner. I updated the PR.